src/app/components/annotations/pages/verification/verification.component.ts

```diff
@@ -246,18 +234,107 @@ class VerificationComponent
     this.hasUnsavedChanges = false;
   }
 
+  protected handleDecision(decisionEvent: Event): void {
+    if (!this.isDecisionEvent(decisionEvent)) {
+      console.error("Received invalid decision event", decisionEvent);
+      return;
+    }
+
+    this.hasUnsavedChanges = true;
+
+    const decision = decisionEvent.detail;
+    for (const [subject, receipt] of decision) {
+      const change = receipt.change;
+
+      if (Object.prototype.hasOwnProperty.call(change, "verification")) {
+        this.handleVerificationDecision(subject);
+      }
+
+      if (Object.prototype.hasOwnProperty.call(change, "newTag")) {
+        this.handleTagCorrectionDecision(subject);
+      }
+    }
+  }
+
+  private handleVerificationDecision(subjectWrapper: SubjectWrapper): void {
+    const subject = subjectWrapper.subject as Readonly<AudioEvent>;
```

Comment:
Aren't our subject's `Annotation`s?

Reply:
Probably need to change something at the `Annotation` level model here.

`Annotation` implements the `IAudioEvent` interface, but does not extend the `AudioEvent` class.
So it has some poor type checking, and is technically incorrect.